edit/lint.js

```diff
@@ -0,0 +1,253 @@
+  editors.forEach((cm, index) => {
+    if (cm.state.lint && cm.state.lint.html) {
+      const html = '<caption>' + label + ' ' + (index + 1) + '</caption>' + cm.state.lint.html;
+      const newBlock = newContent.appendChild(tHTML(html, 'table'));
+
+      newBlock.cm = cm;
+      issueCount += newBlock.rows.length;
+
+      const block = content.children[newContent.children.length - 1];
+      const blockChanged = !block || cm !== block.cm || html !== block.innerHTML;
+      someBlockChanged |= blockChanged;
+      cm.state.lint.reportDisplayed = blockChanged;
+    }
+  });
+  if (someBlockChanged || newContent.children.length !== content.children.length) {
+    document.getElementById('issue-count').textContent = issueCount;
+    container.replaceChild(newContent, content);
+    container.style.display = newContent.children.length ? 'block' : 'none';
+    resizeLintReport();
+  }
+}
+
+function resizeLintReport() {
+  const magicBuffer = 20; // subtracted value to prevent scrollbar
+  const content = $('#lint table');
+  if (content.children.length) {
+    const bounds = content.getBoundingClientRect();
+    const newMaxHeight = bounds.bottom <= window.innerHeight ? '' :
+      // subtract out a bit of padding or the vertical scrollbar extends beyond the viewport
+      (window.innerHeight - bounds.top - magicBuffer) + 'px';
+    if (newMaxHeight !== content.style.maxHeight) {
+      content.parentNode.style.maxHeight = newMaxHeight;
+    }
+  }
+}
+
+function gotoLintIssue(event) {
+  const issue = event.target.closest('tr');
+  if (!issue) {
+    return;
+  }
+  const block = issue.closest('table');
+  makeSectionVisible(block.cm);
+  block.cm.focus();
+  block.cm.setSelection({
+    line: parseInt(issue.querySelector('td[role="line"]').textContent) - 1,
+    ch: parseInt(issue.querySelector('td[role="col"]').textContent) - 1
+  });
+}
+
+function toggleLintReport() {
+  document.getElementById('lint').classList.toggle('collapsed');
+}
+
+function showLintHelp() {
+  let list = '<ul class="rules">';
+  let header = '';
+  if (prefs.get('editor.linter') === 'csslint') {
+    header = t('issuesHelp', '<a href="https://github.com/CSSLint/csslint" target="_blank">CSSLint</a>');
+    list += CSSLint.getRules().map(rule =>
+      '<li><b>' + rule.name + '</b><br>' + rule.desc + '</li>'
+    ).join('');
+  } else {
+    const rules = [];
+    const url = 'https://stylelint.io/user-guide/rules/';
+    header = t('issuesHelp', `<a href="${url}" target="_blank">stylelint</a>`);
+    // to-do: change this to a generator
+    $$('#lint td[role="severity"]').forEach(el => {
+      const rule = el.title.replace('Rule: (', '').replace(/[()]/g, '').trim();
+      if (!rules.includes(rule)) {
+        list += `<li><a target="_blank" href="${url}${rule}/">${rule}</a></li>`;
+        rules.push(rule);
+      }
+    });
+  }
+  return showHelp(t('issues'), header + list + '</ul>');
+}
+
+function setupStylelintSettingsEvents() {
+  let timer;
```

Comment:
Err, my comment above is mostly wrong because the variable is used in the inner function. Well, you can still use the existing `debounce()` function. Also, check if it's possible to close the dialog with an error, reopen the rules config dialog within 3 seconds and still get the previously scheduled error message. I think it's possible and you need to rework the `$('#help-popup .error')` check. Maybe use an expando property: `$('#help-popup .error').hideTimer = setTimeout(.......)` and check/clear it before invoking showHelp.